### PR TITLE
Added Hexadecimal to Binary Conversion

### PR DESCRIPTION
**Pull request by lucasa1212**
### **Describe your change:**
Added hexadecimal to binary conversion

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by kambum285 — APPROVED** [review]

**Review by onlinejudge95 — CHANGES_REQUESTED** [review]
More doc tests covering the paths like empty string provided or paths where the initial string is not a hexadecimal string.

File: conversions/hexadecimal_to_binary.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    Convert hexadecimal string to binary string
+    >>> hex_to_binary("3f")
+    '0b00111111'
+    >>> hex_to_binary("-D2")
+    '-0b11010010'
+    >>> hex_to_binary("Ac")
+    '0b10101100'
+    >>> hex_to_binary("45")
+    '0b01000101'
+    >>> hex_to_binary("")
```
Comment (onlinejudge95) [inline conversions/hexadecimal_to_binary.py:15]: Need to fix the given doctest

**Review by onlinejudge95 — CHANGES_REQUESTED** [review]

**Comment by mrmaxguns** [comment]
Hello! It looks like this algorithm has [already been implemented](https://github.com/TheAlgorithms/Python/blob/master/conversions/hex_to_bin.py). However, the other implementation of the algorithm uses bitwise operators to accomplish the conversion, so this may be OK.

Please also note that pre-commit is failing due to a test failure. Please fix this error by fixing mistakes in the doctests or in the code itself.

**Comment by onlinejudge95** [comment]
Last call, let's wait for 1-2 days then let's close it as the author is not responding
